crates/ide/src/references/rename.rs

```diff
@@ -20,29 +20,66 @@ use crate::{
     SourceChange, SourceFileEdit, TextRange, TextSize,
 };
 
+#[derive(Debug)]
+pub(crate) struct RenameError(pub(crate) String);
+
+impl RenameError {
+    pub(crate) fn new<T: Into<String>>(message: T) -> RenameError {
```

Comment:
This can be private, and I think doesn't need `Into` polymorphism: https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#avoid-monomorphization